app/Http/Controllers/DocumentController.php

```diff
@@ -589,24 +586,9 @@ public function postSponsor($doc)
         $response = null;
 
         if (!isset($sponsor)) {
-            $doc->sponsor()->sync(array());
+            $doc->sponsor()->sync([]);
         } else {
-            switch ($sponsor['type']) {
-                case 'user':
-                    $user = User::find($sponsor['id']);
-                    $doc->userSponsors()->sync(array($user->id));
-                    $doc->groupSponsors()->sync(array());
-                    $response = $user->toArray();
-                    break;
-                case 'group':
-                    $group = Group::find($sponsor['id']);
-                    $doc->groupSponsors()->sync(array($group->id));
-                    $doc->userSponsors()->sync(array());
-                    $response = $group->toArray();
-                    break;
-                default:
-                    throw new Exception('Unknown sponsor type '.$sponsor['type']);
-            }
+            $doc->sponspors()->sync([$sponsor['id']]);
```

Comment:
@sethetter Looks like a typo here.  Also, is there any time where you can actually have *no* sponsor ?  Seems like that should throw an error maybe.